nautobot/core/ui/object_detail.py

```diff
@@ -908,8 +908,15 @@ def get_extra_context(self, context: Context):
 
         try:
             list_route = reverse(get_route_for_model(body_content_table_model, "list"))
-            body_content_table_list_url = f"{list_route}?{related_field_name}={obj.pk}"
         except NoReverseMatch:
+            list_url_attr = getattr(self.table_class, "list_url", "")
+            try:
+                list_route = reverse(list_url_attr)
+            except NoReverseMatch:
+                list_route = None
```

Comment:
This feels backwards to me - if a `list_url` is explicitly specified, shouldn't that take precedence over the `get_route_for_model()` case? Should it be something like:

```py
if not list_url := getattr(self.table_class, "list_url", None):
    list_url = get_route_for_model(body_content_table_model, "list")
try:
    list_route = reverse(list_url)
except NoReverseMatch:
    list_route = None
```